worker/src/lib.rs

```diff
@@ -269,6 +242,90 @@ pub async fn witness_tweet(
     })
 }
 
+#[wasm_bindgen]
+pub async fn witness_discord(
+    secret_key_jwk: String,
+    public_key_tezos: String,
+    discord_authorization_key: String,
+    discord_handle: String,
+    channel_id: String,
+    message_id: String,
+) -> Promise {
+    initialize_logging();
+
+    future_to_promise(async move {
+        let pk: JWK = jserr!(jwk_from_tezos_key(&public_key_tezos));
+        let sk: JWK = jserr!(serde_json::from_str(&secret_key_jwk));
+        let discord_res = jserr!(
+            discord::retrieve_discord_message(discord_authorization_key, channel_id, message_id)
+                .await
+        );
+        let formatted_discord_handle = format!(
+            "{}#{}",
+            discord_res.author.username, discord_res.author.discriminator
+        );
+        let mut vc = jserr!(discord::build_discord_vc(&pk, &formatted_discord_handle));
+
+        // Check for matching handles
+        if discord_handle != formatted_discord_handle {
+            jserr!(Err(anyhow!(format!(
+                "Different Discord handle {} v. {}",
+                discord_handle, formatted_discord_handle
+            ))));
+        }
+
+        let (sig_target, sig) = jserr!(twitter::extract_signature(discord_res.content));
```

Comment:
That function should probably move out of the twitter module to avoid any confusion

Reply:
Refactored to a standalone function in lib.rs